libkbfs/folder_block_ops.go

```diff
@@ -2208,8 +2250,60 @@ func (fbo *folderBlockOps) CleanupSyncState(
 	if df := fbo.dirtyFiles[file.tailPointer()]; df != nil {
 		df.resetSyncingBlocksToDirty()
 	}
+}
+
+// cleanUpUnusedBlocks cleans up the blocks from any previous failed
+// sync attempts.
+func (fbo *folderBlockOps) cleanUpUnusedBlocks(ctx context.Context,
+	md *RootMetadata, syncState fileSyncState, fbm *folderBlockManager) error {
+	numToClean := len(syncState.si.toCleanIfUnused)
+	if numToClean == 0 {
+		return nil
+	}
+
+	// What blocks are referenced in the successful MD?
+	refs := make(map[BlockPointer]bool)
+	for _, op := range md.data.Changes.Ops {
+		for _, ptr := range op.Refs() {
+			if ptr != zeroPtr {
```

Comment:
can refs be zero, too? From the code above, doesn't seem like it. Panic instead?